src/python/pants/engine/internals/graph_test.py

```diff
@@ -1100,7 +1100,7 @@ def assert_dependencies_resolved(
         enable_dep_inference: bool = False,
     ) -> None:
         target = self.request_single_product(WrappedTarget, requested_address).target
-        args = ["--dependency-inference"] if enable_dep_inference else []
+        args = ["--python-infer-imports"] if enable_dep_inference else []
```

Comment:
This can be removed now that inference is always used.

Reply:
It's not: `--infer-imports` is off by default.

Reply:
This is `engine/graph_test.py`. It doesn't know that the Python subsystem exists. My point is that it's now up to the dep inference rule whether it runs or not; `graph.py` will run everything possible now. In this unit test, this means that the demo inference rule will always run and this option is irrelevant.

Reply:
Got it!